src/object.js

```diff
@@ -165,11 +176,16 @@ function defineProp(object, name, opts = {}) {
     (object.__deps__[prop] = object.__deps__[prop] || []).push(name);
 
     if (prop.indexOf('.') !== -1) {
-      let segments = prop.split('.'), first = segments[0];
+      let segments = prop.split('.'), first = segments[0], last = segments[1];
```

Comment:
is there a reason for assigning first and last before seeing if the segments array is longer than 2?

Reply:
Its really just a stylistic preference to initialize variables where they are defined. I suppose we're paying a small performance cost when the length is greater than 2, but thats an error case that would prevent the code from running.